src/BootApp.ts

```diff
@@ -8,6 +8,15 @@ import { ClassPathData, MainClassData } from "./types/jdtls";
 import { isActuatorJarFile, isAlive } from "./utils";
 import { BootAppItem } from "./views/items/BootAppItem";
 import * as lsp from "vscode-languageclient";
+import * as async from "async";
+
+const searchQueue = async.queue(async (path: string, callback: async.AsyncResultCallback<string[], Error>) => {
+    try {
+        callback(undefined, await vscode.commands.executeCommand('java.execute.workspaceCommand', 'vscode.java.resolveMainClass', path) as string[]);
+    } catch (error) {
+        callback(error);
+    }
+}, 5);
```

Comment:
I see it uses a hard code number 5 for concurrency, this still likely uses all threads in common thread pool. Since the threshold of Java common thread pool is related with CPU cores, can we limit the max search concurrency related with CPU cores, e.g. cpus/2?

Reply:
Yeah, absolutely.